docs/release-notes/lang-configuration-setting.rst

```diff
@@ -0,0 +1,17 @@
+:Type: Configuration
+:Summary: Add locale setting to configuration (#INSTALLER-3472)
+
+:Description:
+    New Localization section with lang setting is added to Anaconda
+    configuration.
+
+    The setting corresponds to the kickstart lang option.
+
+    Intended use: taking over configuration provided by other tools in Anaconda
+    configuration drop directory (/etc/anaconda/conf.d), eg Gnome Inital Setup.
+
+    Priority of the configuration:
+    inst.lang > kickstart lang > conf.localization.lang > geolocation
```

Comment:
This is stressing me out. The configuration file always has the lowest priority, because it provides defaults. It is documented here: https://anaconda-installer.readthedocs.io/en/latest/configuration-files.html#runtime-configuration-file So it cannot be never ever higher than a kickstart file or boot options. I strongly recommend to follow this practice. You already struggle with mixed priorities of kickstart commands and boot options.

I looked up the Jira issue and I don't fully understand the use cases you are trying to cover. Don't you target Fedora Workstation? Why are kickstart and boot options dragged into this? The comment at https://github.com/rhinstaller/anaconda-webui/issues/1#issuecomment-1516406850 says that the pre-selected language will be applied to the Live environment. Anaconda should be actually able to pick up this value automatically if the language is not set:

https://github.com/rhinstaller/anaconda/blob/ea679e9aa9d5bee788c918e4e61ba208bb23b1d6/pyanaconda/localization.py#L647-L656

Btw. the language seems to be a good candidate for introducing someting like `SetLanguageWithPriority` (see `SetTimezoneWithPriority`) and get rid of `LanguageKickstarted`, but that is probably out of scope of this issue.

Reply:
> This is stressing me out. The configuration file always has the lowest priority, because it provides defaults. It is documented here: https://anaconda-installer.readthedocs.io/en/latest/configuration-files.html#runtime-configuration-file So it cannot be never ever higher than a kickstart file or boot options. I strongly recommend to follow this practice. You already struggle with mixed priorities of kickstart commands and boot options.

I am not following, the priority scheme wanted to say that the priority is the same as before with the configuration option just overriding the default or geolocation.
But it really seems we are misusing configuration to do something it should not to. We should then rather turn off geolocation for our use case.

Reply:
> I am not following, the priority scheme wanted to say that the priority is the same as before with the configuration option just overriding the default or geolocation.

Right, I have got confused by the order. My mistake. Still, it doesn't make sense to prioritize the configuration option over the geolocation result.

> We should then rather turn off geolocation for our use case.

Agreed.